src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/LearnersEducationService.kt

```diff
@@ -0,0 +1,77 @@
+package uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.service
+
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.config.NoDataWithCodeFoundException
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.config.ResourceNotFoundException
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.data.LearnersCourse
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.data.LearnersCourseList
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.data.curiousapi.LearnersEducationList
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.jpa.repository.PrisonerRepository
+import uk.gov.justice.digital.hmpps.hmppsresettlementpassportapi.service.external.CuriousApiService
+
+@Service
+class LearnersEducationService(
+  val learnersEducationApiService: CuriousApiService,
+  private val prisonerRepository: PrisonerRepository,
+) {
+  fun getLearnersEducationCourseData(nomsId: String, pageNumber: Int, pageSize: Int): LearnersCourseList {
+    if (nomsId.isBlank() || nomsId.isEmpty()) {
+      throw NoDataWithCodeFoundException("Prisoners", nomsId)
+    }
+
+    val prisoner = prisonerRepository.findByNomsId(nomsId)
+      ?: throw ResourceNotFoundException(
+        "Prisoner with id $nomsId not found in the DB",
+      )
+
+    if (pageNumber < 0 || pageSize <= 0) {
+      throw NoDataWithCodeFoundException(
+        "Data",
+        "Page $pageNumber and Size $pageSize",
+      )
+    }
+
+    val courses = learnersEducationApiService.getLearnersEducation(prisoner.nomsId)
+
+    val startIndex = (pageNumber * pageSize)
+    if (startIndex >= courses.content?.size!!) {
+      throw NoDataWithCodeFoundException(
+        "Data",
+        "Page $pageNumber",
+      )
+    }
+    val courseList = objectMapper(courses)
+
+    val endIndex = (pageNumber * pageSize) + (pageSize)
+    if (startIndex < endIndex && endIndex <= courseList.size) {
+      val cList = courseList.subList(startIndex, endIndex)
+      return LearnersCourseList(cList, cList.toList().size, pageNumber, courseList.size, (endIndex == courses.size))
```

Comment:
If I am understanding correctly we are pulling all the data from their paged api and then applying our own paging to it manually in memory? 

I don't understand why we'd not just pass the page fields through to the underlying api

Reply:
Done